Makefile

```diff
@@ -13,12 +13,15 @@ MIN_GOLANG_VERSION=1.16 # for building ghpc
 ENG = ./cmd/... ./pkg/...
 TERRAFORM_FOLDERS=$(shell find ./modules ./community/modules ./tools -type f -name "*.tf" -not -path '*/\.*' -exec dirname "{}" \; | sort -u)
 PACKER_FOLDERS=$(shell find ./modules ./community/modules ./tools -type f -name "*.pkr.hcl" -not -path '*/\.*' -exec dirname "{}" \; | sort -u)
+GIT_TAG_VERSION=$(shell git tag --points-at HEAD)
+GIT_BRANCH=$(shell git branch --show-current)
+GIT_COMMIT_INFO=$(shell git describe --tags --dirty --long)
 
 # RULES MEANT TO BE USED DIRECTLY
 
 ghpc: warn-go-version warn-terraform-version warn-packer-version $(shell find ./cmd ./pkg ghpc.go -type f)
 	$(info **************** building ghpc ************************)
-	go build ghpc.go
+	@go build -ldflags="-X 'main.gitTagVersion=$(GIT_TAG_VERSION)' -X 'main.gitBranch=$(GIT_BRANCH)' -X 'main.gitCommitInfo=$(GIT_COMMIT_INFO)'" ghpc.go
```

Comment:
I tried making when not in a git repo and the output from make looks like:
```
~/dev/test-zip$ make
**************** building ghpc ************************
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
```

In this case the binary still built and it correctly fell back to the in text version, but if we are considering this a valid use case we probably shouldn't be printing fatal when everything is actually fine. Can we suppress this specific warning, or perhaps tell the user that they are not building from a git repository. 

I also want to make sure we don't suppress all output since there could be something else useful. Perhaps we could first check if git is available and then have a different build command based on yes or now.

Steps to repro:
- checkout your branch
- git archive HEAD --format=zip -o toolkit-zip.zip
- cd ..
- mkdir test-zip
- mv hpc-toolkit/toolkit-zip.zip test-zip/
- cd test-zip/
- unzip toolkit-zip.zip
- make

Reply:
correction, this is the case that I have git installed but I am not in a git repo.